File: mercado_backend/src/main/java/com/mercado/pagos/servicios/service/SocioService.java
package com.mercado.pagos.servicios.service;

import com.mercado.pagos.servicios.dto.request.SocioRequestDTO;
import com.mercado.pagos.servicios.dto.response.PageResponseDTO;
import com.mercado.pagos.servicios.dto.response.SocioResponseDTO;

public interface SocioService {

    SocioResponseDTO crearSocio(SocioRequestDTO requestDTO);

    PageResponseDTO<SocioResponseDTO> listarSocios(int page, int size);

    PageResponseDTO<SocioResponseDTO> buscarSocios(String termino, int page, int size);

    SocioResponseDTO obtenerSocioPorId(Long id);

    SocioResponseDTO actualizarSocio(Long id, SocioRequestDTO requestDTO);

    SocioResponseDTO cambiarEstadoSocio(Long id, String estado);

}

File: mercado_frontend/src/app/features/socios/services/socio.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';
import { EstadoSocio, PageResponseDTO, SocioRequestDTO, SocioResponseDTO } from '../models/socio.models';

@Injectable({
  providedIn: 'root'
})
export class SocioService {
  private readonly api = inject(ApiService);
  private readonly basePath = '/socios';

  listar(page: number, size: number): Observable<PageResponseDTO<SocioResponseDTO>> {
    return this.api.get<PageResponseDTO<SocioResponseDTO>>(this.basePath, { page, size });
  }

  buscar(termino: string, page: number, size: number): Observable<PageResponseDTO<SocioResponseDTO>> {
    return this.api.get<PageResponseDTO<SocioResponseDTO>>(`${this.basePath}/buscar`, { termino, page, size });
  }

  obtenerPorId(id: number): Observable<SocioResponseDTO> {
    return this.api.get<SocioResponseDTO>(`${this.basePath}/${id}`);
  }

  crear(request: SocioRequestDTO): Observable<SocioResponseDTO> {
    return this.api.post<SocioResponseDTO, SocioRequestDTO>(this.basePath, request);
  }

  actualizar(id: number, request: SocioRequestDTO): Observable<SocioResponseDTO> {
    return this.api.put<SocioResponseDTO, SocioRequestDTO>(`${this.basePath}/${id}`, request);
  }

  cambiarEstado(id: number, estado: EstadoSocio): Observable<SocioResponseDTO> {
    return this.api.patch<SocioResponseDTO, null>(`${this.basePath}/${id}/estado?estado=${estado}`, null);
  }
}
